File: plantwatcher/src/main/java/com/example/plantwatcher/models/Configs.java
package com.example.plantwatcher.models;

import jakarta.persistence.*;

@Entity
@Table(name = "Configs")
public class Configs {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "moisture_starter")
    private Integer moistureStarter;

    @Column(name = "temperature_starter")
    private int temperatureStarter;

    @Column(name = "insolation_starter")
    private int insolationStarter;

    public Configs() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getMoistureStarter() {
        return moistureStarter;
    }

    public void setMoistureStarter(Integer moistureStarter) {
        this.moistureStarter = moistureStarter;
    }

    public int getTemperatureStarter() {
        return temperatureStarter;
    }

    public void setTemperatureStarter(int temperatureStarter) {
        this.temperatureStarter = temperatureStarter;
    }

    public int getInsolationStarter() {
        return insolationStarter;
    }

    public void setInsolationStarter(int insolationStarter) {
        this.insolationStarter = insolationStarter;
    }

    // Getters and setters
}

File: PlanterUi/src/app/configs/model/configs.ts
export class Configs {
    public name: string;
    public moistureStarter: number;
    public insolationStarter: number;
    public temperatureStarter: number;
  

    constructor(name: string, moistureStarter: number, insolationStarter: number, temperatureStarter: number) {
      this.name = name;
      this.moistureStarter = moistureStarter;
      this.insolationStarter = insolationStarter;
      this.temperatureStarter = temperatureStarter;
    }
    
    // getters and setters
    public getName(): string {
      return this.name;
    }
    
    public setName(name: string): void {
      this.name = name;
    }
    
    public getMoistureStarter(): number {
      return this.moistureStarter;
    }
    
    public setMoistureStarter(moistureStarter: number): void {
      this.moistureStarter = moistureStarter;
    }
    
    public getInsolationStarter(): number {
      return this.insolationStarter;
    }
    
    public setInsolationStarter(insolationStarter: number): void {
      this.insolationStarter = insolationStarter;
    }
    
    public getTemperatureStarter(): number {
      return this.temperatureStarter;
    }
    
    public setTemperatureStarter(temperatureStarter: number): void {
      this.temperatureStarter = temperatureStarter;
    }
  }
